pkg/system/phase2_creating.go

```diff
@@ -171,6 +176,101 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	return nil
 }
 
+func (r *Reconciler) UpgradeDbPostgres16() error {
+	var err error = nil
+
+	// Check the desired image, if it is not as same as db image to upgrade,
+	// don't consider for upgrade
+	// Note: Intention of the below call is to fetch the desired image from Noobaa CR or env var
+	// Passing empty Arg "" to GetDesiredDBImage() in the below case
```

Comment:
As a first condition, check the postgres major version

Reply:
Sure

Reply:
The check is added in `isPostgresUpgradeReqd()`